### PR TITLE
[LF] Cleanup API for disclosed contract

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
In particular drop info the engine do not care about.

companion PR https://github.com/DACH-NY/canton/pull/13232

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ if [[ "$(uname)" == "Darwin" ]]; then
   tag_filter="$tag_filter,-dont-run-on-darwin,-scaladoc,-pdfdocs"
 fi
 
-SKIP_DEV_CANTON_TESTS=false
+SKIP_DEV_CANTON_TESTS=true
```
Comment (remyhaemmerle-da) [inline build.sh:21]: The change break our patched canton.

File: daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/MultiParticipantIT.scala [hunk 2]
```diff
@@ -22,7 +22,6 @@ final class MultiParticipantIT
   val dar = stableDar
 
   final override protected lazy val darFiles = List(stableDarPath)
-  final override protected lazy val devMode = true
```
Comment (remyhaemmerle-da) [inline daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/MultiParticipantIT.scala:25]: Not related to this PR, but there is no reason to run that in dev mode

File: build.sh (hunk 1 above)
Comment (remyhaemmerle-da) [reply build.sh:21]: cc @nickchapman-da @moisesackerman-da

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala [hunk 3]
```diff
@@ -691,7 +691,7 @@ object Transaction {
       dependsOnTime: Boolean,
       nodeSeeds: ImmArray[(NodeId, crypto.Hash)],
       globalKeyMapping: Map[GlobalKey, Option[Value.ContractId]],
-      processedDisclosedContracts: ImmArray[ProcessedDisclosedContract],
+      diclosedEvents: ImmArray[Node.Create],
```
Comment (akrmn) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala:694]: typo 
```suggestion
      disclosedEvents: ImmArray[Node.Create],
```
(plus a few others)

**Review by akrmn — APPROVED** [review]
LGTM, thanks @remyhaemmerle-da!

**Review by andreaslochbihler-da — APPROVED** [review]

**Review by carlpulley-da — APPROVED** [review]

**Pull request merged**